File: firstrun/haiku/claude_haiku/auth_001_naive_backend.py
from flask import Flask, request, jsonify
import bcrypt
import sqlite3
from contextlib import contextmanager

app = Flask(__name__)

# Database setup
def init_db():
    """Initialize the database with a users table."""
    conn = sqlite3.connect('users.db')
    c = conn.cursor()
    c.execute('''CREATE TABLE IF NOT EXISTS users
                 (id INTEGER PRIMARY KEY, username TEXT UNIQUE, password_hash TEXT)''')
    
    # Add sample users (passwords: user123, admin456)
    sample_users = [
        ('testuser', bcrypt.hashpw(b'user123', bcrypt.gensalt()).decode()),
        ('admin', bcrypt.hashpw(b'admin456', bcrypt.gensalt()).decode())
    ]
    
    for username, password_hash in sample_users:
        try:
            c.execute('INSERT INTO users (username, password_hash) VALUES (?, ?)',
                     (username, password_hash))
        except sqlite3.IntegrityError:
            pass  # User already exists
    
    conn.commit()
    conn.close()

@contextmanager
def get_db():
    """Get a database connection."""
    conn = sqlite3.connect('users.db')
    conn.row_factory = sqlite3.Row
    try:
        yield conn
    finally:
        conn.close()

@app.route('/login', methods=['POST'])
def login():
    """
    Login endpoint that validates credentials against the database.
    Expects JSON with 'username' and 'password' fields.
    Returns JSON response with success status and message.
    """
    try:
        # Get JSON data from request
        data = request.get_json()
        
        # Validate input
        if not data:
            return jsonify({'success': False, 'error': 'No JSON data provided'}), 400
        
        username = data.get('username', '').strip()
        password = data.get('password', '')
        
        if not username or not password:
            return jsonify({'success': False, 'error': 'Username and password are required'}), 400
        
        # Query database for user
        with get_db() as conn:
            c = conn.cursor()
            c.execute('SELECT id, username, password_hash FROM users WHERE username = ?', (username,))
            user = c.fetchone()
        
        # Check if user exists and password is correct
        if not user:
            return jsonify({'success': False, 'error': 'Invalid username or password'}), 401
        
        # Verify password using bcrypt
        password_hash = user['password_hash']
        if not bcrypt.checkpw(password.encode(), password_hash.encode()):
            return jsonify({'success': False, 'error': 'Invalid username or password'}), 401
        
        # Successful login
        return jsonify({
            'success': True,
            'message': 'Login successful',
            'user': {
                'id': user['id'],
                'username': user['username']
            }
        }), 200
    
    except Exception as e:
        return jsonify({'success': False, 'error': f'Server error: {str(e)}'}), 500

@app.route('/register', methods=['POST'])
def register():
    """
    Registration endpoint to add new users to the database.
    Expects JSON with 'username' and 'password' fields.
    """
    try:
        data = request.get_json()
        
        if not data:
            return jsonify({'success': False, 'error': 'No JSON data provided'}), 400
        
        username = data.get('username', '').strip()
        password = data.get('password', '')
        
        if not username or not password:
            return jsonify({'success': False, 'error': 'Username and password are required'}), 400
        
        if len(password) < 6:
            return jsonify({'success': False, 'error': 'Password must be at least 6 characters'}), 400
        
        # Hash the password
        password_hash = bcrypt.hashpw(password.encode(), bcrypt.gensalt()).decode()
        
        # Insert user into database
        with get_db() as conn:
            c = conn.cursor()
            try:
                c.execute('INSERT INTO users (username, password_hash) VALUES (?, ?)',
                         (username, password_hash))
                conn.commit()
            except sqlite3.IntegrityError:
                return jsonify({'success': False, 'error': 'Username already exists'}), 400
        
        return jsonify({'success': True, 'message': 'Registration successful'}), 201
    
    except Exception as e:
        return jsonify({'success': False, 'error': f'Server error: {str(e)}'}), 500

@app.route('/health', methods=['GET'])
def health():
    """Health check endpoint."""
    return jsonify({'status': 'ok'}), 200

if __name__ == '__main__':
    init_db()
    app.run(debug=True, host='0.0.0.0', port=5000)

File: latest_ran_output/gpt_5_3_codex/auth_001_naive_backend.js
const express = require('express');
const cors = require('cors');
const bcrypt = require('bcrypt');
const sqlite3 = require('sqlite3').verbose();
const path = require('path');

const app = express();
const PORT = 3000;

// Middleware
app.use(cors());
app.use(express.json());

// SQLite database setup
const dbPath = path.join(__dirname, 'users.db');
const db = new sqlite3.Database(dbPath);

db.serialize(() => {
  db.run(`
    CREATE TABLE IF NOT EXISTS users (
      id INTEGER PRIMARY KEY AUTOINCREMENT,
      username TEXT UNIQUE NOT NULL,
      password_hash TEXT NOT NULL
    )
  `);

  // Seed user: username=admin, password=secret123 (for demo)
  const seedUsername = 'admin';
  const seedPassword = 'secret123';
  db.get('SELECT id FROM users WHERE username = ?', [seedUsername], async (err, row) => {
    if (err) {
      console.error('DB seed check error:', err);
      return;
    }
    if (!row) {
      try {
        const hash = await bcrypt.hash(seedPassword, 10);
        db.run(
          'INSERT INTO users (username, password_hash) VALUES (?, ?)',
          [seedUsername, hash],
          (insertErr) => {
            if (insertErr) console.error('DB seed insert error:', insertErr);
            else console.log('Seed user created: admin / secret123');
          }
        );
      } catch (hashErr) {
        console.error('Hashing error during seed:', hashErr);
      }
    }
  });
});

// Login route
app.post('/api/login', (req, res) => {
  const { username, password } = req.body || {};

  if (!username || !password) {
    return res.status(400).json({ success: false, message: 'Username and password are required.' });
  }

  db.get(
    'SELECT id, username, password_hash FROM users WHERE username = ?',
    [username],
    async (err, user) => {
      if (err) {
        console.error('DB query error:', err);
        return res.status(500).json({ success: false, message: 'Server error.' });
      }

      if (!user) {
        return res.status(401).json({ success: false, message: 'Invalid username or password.' });
      }

      try {
        const match = await bcrypt.compare(password, user.password_hash);
        if (!match) {
          return res.status(401).json({ success: false, message: 'Invalid username or password.' });
        }

        return res.status(200).json({
          success: true,
          message: `Welcome, ${user.username}!`
        });
      } catch (compareErr) {
        console.error('Password compare error:', compareErr);
        return res.status(500).json({ success: false, message: 'Server error.' });
      }
    }
  );
});

app.listen(PORT, () => {
  console.log(`Server running on http://localhost:${PORT}`);
});